StorageUtility/Linux/StorageUtility.cpp

```diff
@@ -15,253 +15,185 @@ See the License for the specific language governing permissions and
 limitations under the License.
 </License>
 */
-
+#include <iostream>
 #include <libudev.h>
+#include <fcntl.h>
+#include <map>
 #include <memory>
 #include <string.h>
+#include <unistd.h>
 
-#include "StorageUtility.h"
 #include "Device.h"
+#include "StorageUtility.h"
 
 namespace vtStor
 {
-
-eErrorCode GetStorageDevices(std::vector<std::shared_ptr<cDeviceInterface>>& Devices, eOnErrorBehavior OnErrorBehavior)
+struct s_MappingSystemPathAndDeviceHandle
 {
-    eErrorCode error = eErrorCode::None;
+    std::map<String, vtStor::sStorageAdapterProperty> s_MappingSystemPathToProperties;
+    std::map<DeviceHandle, vtStor::String>  s_MappingDeviceHandleToSysPath;
+};
 
-    sEnumerateDevicesCallback callBack;
-    callBack.Data = (void*)&Devices;
+std::unique_ptr<s_MappingSystemPathAndDeviceHandle>  MappingSystemPathAndDeviceHandle;
```

Comment:
When maps were used then we reset unique point with nullptr (in GetStorageAdapterProperty function )